encode.go

```diff
@@ -219,24 +226,28 @@ func encodeValue(buf []byte, prefix, name string, fv reflect.Value, inArray, arr
 
 func appendDocInline(buf []byte, doc string) []byte {
 	if doc != "" {
-		return append(append(append(append(buf, ' '), '#'), ' '), doc...)
+		return append(append(buf, ' ', '#', ' '), doc...)
 	} else {
 		return buf
 	}
 }
 
 func appendDoc(buf []byte, doc string) []byte {
-	return appendDocInline(buf, doc)[1:]
+	return append(buf, appendDocInline(nil, doc)[1:]...)
```

Comment:
@timraymondFYI this was the fix for the docs encoding. Previously it would the first character of the entire buffer which was rarely the the added space from appendDocInline.